### ISSUE TITLE
Persistent black screen when trying to view/control remote desktop

### ISSUE THREAD

**Issue by albadedan** [Ylianst/MeshCentral#5503]
**Describe the bug**
I am unable to properly control any of my remote devices.

Upon connecting to the device, the status does get to 'Connected', and end users report that my mouse movements are sent to their device, but I cannot see their desktop.

The black screen bug seems to appear in this issue list a lot, such as #3916 #4397 #4554 #4668 and #5025, among others.

Most of the solutions to those issues revolve around disabling WebRTC and one of them around setting minify to false, but neither of those solutions have worked for me.

All in all, I have tried:
- Restarting the docker container
- Updating the image from 1.1.5 to 1.1.15
- Sending 'service restart' in the device console, as here: https://github.com/Ylianst/MeshCentral/issues/4554#issuecomment-1253255956
- Performing 'Clear the core', as here:  https://github.com/Ylianst/MeshCentral/issues/4554#issuecomment-1253148927
- Performing 'Upload default server core'
- Setting WebRTC to false, as here: https://github.com/Ylianst/MeshCentral/issues/4397#issuecomment-1215336931
- Setting WebRTC to true
- Setting minify to true
- Setting minify to false, as here: https://github.com/Ylianst/MeshCentral/issues/4668#issuecomment-1290550217

Nothing has fixed the problem.

To be honest, this issue first occurred for me about a month or two ago, but I was busy elsewhere.
Originally, everything worked fine, but one day they all just stopped working.

Some of the devices have always-on VPNs, but some don't. Some are local to the host server, some are remote over WAN. All are running Windows 11 Pro, in various states of update.

What else can I do?

**To Reproduce**
1. Select any device
2. Click 'Desktop' > click 'Connect' (ensure 'input' checkbox is ticked)

**Expected behavior**
See and control remote device.

**Server Software (please complete the following information):**
 - OS: Ubuntu
 - Virtualization :Docker
 - Network: cloudflare tunnel
 - Version: 1.1.15

**Client Device (please complete the following information):**
 - Device: Laptop
 - OS: Win 11 Pro
 - Network: Local to Meshcentral
 - Browser: Google Chrome, Firefox and Edge

**Remote Device (please complete the following information):**
 - Device: All devices, mixture of desktops and laptops, all Win 11 Pro
 - OS: Various states of update, but all Win 11 Pro
 - Network: Local to Meshcentral and Remote over WAN

**Your config.json file**
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "plugins":{"enabled": false},
    "_mongoDb": null,
    "cert": "domain",
    "_WANonly": true,
    "_LANonly": true,
    "sessionKey": "some key",
    "port": 443,
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": false,
    "SelfUpdate": true,
    "AllowFraming": false,
    "WebRTC": true
  },
  "domains": {
    "": {
      "title": "Mesh",
      "_title2": "Servername",
      "minify": false,
      "NewAccounts": false,
      "localSessionRecording": false,
      "_userNameIsEmail": true,
      "certUrl": "domain"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "myemail@mydomain.com",
    "_names": "myserver.mydomain.com",
    "production": false
  }
}

```

**Comment by si458** [comment]
you say you use 'cloudflare tunnel' 
please refer to #5302
cloudflare DNS proxy and Tunnels is BROKEN at the moment
sadly we need to upgrade the agents to comply with the websocket standard, which isnt an easy job
the only thing you can do is use direct access (disable the proxy option)

**Comment by albadedan** [comment]
Hi Si, thanks for getting back to me.
That's an interesting issue you've linked, though the symptoms are not quite the same.
I don't get this, for example: https://github.com/Ylianst/MeshCentral/issues/5302#issuecomment-1692930569
But until now, I had not bothered to try to gain terminal or file access, and indeed they do not work either.

If I remove the cloudflare integration, does that mean I have to re-create and re-install the agent on all the remote devices? As I understand it, the server information is embedded within the agent exe. Or is it enough to maintain the path, and the existing agents will be happy as long as I'm not using cloudflare?

**Comment by si458** [comment]
in theory (only my trial and error)

you can disable the cloudflare proxy so its points straight to your server
then you can enable `letsencrypt` section in config.json with the same cert name as your dns name,
set `production: false` in letsencrypt to check first as it might take a few mins to get an invalid cert but thats ok as it should show the letsencrypt staging cert, then set `production: true` wait a few mins again, and then should be valid!

the agent certs dont change so you should still be able to access the remote agents no problems
so long as the agents point to the same dns name!

hope that makes sense!

**Comment by albadedan** [comment]
If that can be done, then it needs more skill than I possess. In attempting to put it back to how it was before, I can now only access it via HTTP. Accessing via the previous tunnel route now only results in a websocket error.

Seeing as my environment is entirely windows, I'm just going to have to revert to MSRA and QuickAssist for the time being. 

When I get some time I'll probably just flatten it and start entirely again without any prior cloudflare complications. Gives me a good reason to call everybody in and make sure their machines are up-to-date at the same time.

Appreciate the help though.

**Comment by david85091** [comment]
has this been solved yet?

**Comment by si458** [comment]
@david85091 the cloudflare issue hasnt been resolved still, follow #5302 for updates
